libs/go/zmscli/domain.go

```diff
@@ -38,13 +39,95 @@ func (cli Zms) DeleteDomain(dn string) (*string, error) {
 		}
 		if err == nil {
 			s := "[Deleted domain " + dn + "]"
-			return cli.switchOverFormats(s)
+			message := SuccessMessage{
+				Status:  200,
+				Message: s,
+			}
+			return cli.dumpByFormat(message, cli.buildYAMLOutput)
 		}
 	}
 	return nil, err
 }
 
 func (cli Zms) ImportDomain(dn string, filename string, admins []string) (*string, error) {
+
+	if cli.OutputFormat == JSONOutputFormat || cli.OutputFormat == YAMLOutputFormat {
+		return cli.ImportDomainNew(dn, filename, admins, true)
```

Comment:
The import domain/update domain / export domain didn't work properly for Json (The json only included the "domain" responses without "roles / policies / services").
For these commands, I split the functions into "Old" versions and "New" versions. The "Old" versions are only used for "manual yaml" and will be removed once we remove the "manual yaml" functionality.